### PR TITLE
Create Permissions sample application

### PR DESCRIPTION
**Pull request by KovalevAndrey**
**Description**: Create Permissions sample application by extracting some part of code from sandbox app and simplify things  by removing MVI, RxJava, etc. dependencies 

Screenshot_20210430-143452

### CODE REVIEW

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                            .events(this)
+                            .observe { event ->
+                                if (event.requestCode == REQUEST_CODE_CAMERA) {
+                                    when (event) {
+                                        is RequestPermissionsResult -> view.setText("Permission event: $event")
+                                        is Cancelled -> view.setText("Permission request cancelled")
+                                        else -> Unit
+                                    }
+                                }
+                            }
```
Comment (zsoltk) [inline samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt:44]: You could extract this block, it's a bit too much here

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+interface PermissionsPresenter : RequestCodeClient {
+
+    fun onRequestPermissionsClicked()
+
+    fun onCheckPermissionsClicked()
+}
+
+class PermissionsPresenterImpl(
+        private val permissionRequester: PermissionRequester,
```
Comment (zsoltk) [inline samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt:23]: Could you please set continuation indent to 4 spaces and reformat all files

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsRib.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.badoo.ribs.samples.permissions.rib
+
+import com.badoo.ribs.android.permissionrequester.CanProvidePermissionRequester
+import com.badoo.ribs.core.Rib
+
+interface PermissionsRib : Rib {
```
Comment (zsoltk) [inline samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsRib.kt:6]:
```suggestion
interface PermissionsSample : Rib {
```

No need to mention Rib in the actual name. (Also, you can use the plugin to rename every file in a Rib automatically, no need to do one by one)

File: settings.gradle [hunk 4]
```diff
@@ -35,6 +35,7 @@ include ':samples:dialogs'
 include ':samples:communication-between-nodes-1'
 include ':samples:hello-world'
 include ':samples:plugins'
+include ':samples:android-integration:permissions'
```
Comment (zsoltk) [inline settings.gradle:38]: Can you please keep them organised by alphabetical order, thanks.

**Review by zsoltk — CHANGES_REQUESTED** [review]

**Comment by KovalevAndrey** [comment]
I have fixed the comments first and now extending the functional to having a child RIB that will be receiving permissions callback

**Comment by KovalevAndrey** [comment]
> I have fixed the comments first and now extending the functional to having a child RIB that will be receiving permissions callback

Created child RIB that uses permission while parent RIB doesn't

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt (hunk 1 above)
Comment (zsoltk) [reply samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt]: You resolved but it remained the same :)

File: settings.gradle (hunk 4 above)
Comment (zsoltk) [reply settings.gradle]: Same here.

Comment (zsoltk) [reply settings.gradle]: My bad.

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt (hunk 1 above)
Comment (zsoltk) [reply samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/PermissionsPresenter.kt]: My bad!

File: samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/parent/PermissionsSampleParentView.kt [hunk 5]
```diff
@@ -32,5 +32,10 @@ class PermissionsSampleParentViewImpl private constructor(
         }
     }
 
-    override fun getParentViewForSubtree(subtreeOf: Node<*>): ViewGroup = childContainer
+    override fun getParentViewForSubtree(subtreeOf: Node<*>): ViewGroup =
+        when (subtreeOf.parent?.children?.indexOf(subtreeOf) ?: -1) {
+            0 -> child1Container
+            1 -> child2Container
+            else -> super.getParentViewForSubtree(subtreeOf)
```
Comment (zsoltk) [inline samples/android-integration/permissions/src/main/java/com/badoo/ribs/samples/permissions/rib/parent/PermissionsSampleParentView.kt:39]: This is super hacky :) Normally we should simply rely on incoming interface, but yes, now they are the same. Let's try to figure out something for this.

**Review by zsoltk — APPROVED** [review]

**Pull request merged**